### PR TITLE
create dataset for the selection pass

### PR DESCRIPTION
**Pull request by SzymonKozl**

### CODE REVIEW

**Comment by SzymonKozl** [comment]
New datasets:
* `zpp-murmuras/coupon_select_big_json`: coupon selection with json-encoded structure of xml
* `zpp-murmuras/coupon_select_big_plain`: coupon selection on plain text

File: src/bert_dataset_generation/generate_coupon_selection_ds.py [hunk 1]
```diff
@@ -318,34 +317,48 @@ def json_to_labeled_tokens(data: List[TreeNode], indent: Optional[int] = None) -
...
+
+        # Initial train/test split (80% train, 20% temp)
+        train_texts, temp_texts, train_labels, temp_labels = train_test_split(
+            texts, labels, test_size=0.2, random_state=42
+        )
+
+        # Split temp set into validation (10%) and test (10%)
+        val_texts, test_texts, val_labels, test_labels = train_test_split(
+            temp_texts, temp_labels, test_size=0.5, random_state=42
+        )
```
Comment (GustawB) [inline src/bert_dataset_generation/generate_coupon_selection_ds.py:341]: I think we agreed that we don't necessary need validation. So unless I'm mistaken, I think you can remove this part.

File: src/bert_dataset_generation/readme.md [hunk 2]
```diff
@@ -3,10 +3,10 @@ This directory contains a script that can be used for creating and publishing da
 The script works on a pair of a coupon frame and a content frame. It finds coupons from the first file in the second file.<br/>
 Running the script:<br/>
 ```bash
-python generate_coupon_selection_ds.py <config_path> <ds_name> <create_repo>
+python generate_coupon_selection_ds.py <config_path> <ds_name> <create_repo> <custom_split>
 ```
 Where `config_path` is a path to the configuration file in format demonstrated by `example_config.json` file, and
-create_repo is either 'y' or 'n'; it is used to mark whetver the dataset is being pushed to an existing or new repo. <br/>
+create_repo is either 'y' or 'n'; it is used to mark whether the dataset is being pushed to an existing or new repo. `custom_split` indicates if frames should be divided into splits with the usage of `"split"` fields from config file. It should be either `y` or `n`.<br/>
```
Comment (GustawB) [inline src/bert_dataset_generation/readme.md:9]: Can you elaborate on this?

Comment (GustawB) [reply src/bert_dataset_generation/readme.md:9]: For now in comment
I want to understand this to know whether we need to extend the description in readme

File: src/bert_dataset_generation/generate_coupon_selection_ds.py [hunk 3]
```diff
@@ -240,14 +240,14 @@ def frame_to_json(frame: pd.DataFrame, coupons_frame: pd.DataFrame, fmt: int = 1
         if tree is not None:
             res.append(tree)
         else:
-            res.append({"text": None, "children": {}, "is_coupon": False})
+            res.append({"text": None, "children": {}, "is_coupon": LBL_UNK})
```
Comment (GustawB) [inline src/bert_dataset_generation/generate_coupon_selection_ds.py:243]: What does LBL_UNK represent?

**Review by GustawB — CHANGES_REQUESTED** [review]
NGL config itself starts to look ugly; Can't we simplify it to just take a folder with data and iterate over it? I did something similar in #121 ; maybe you can do something similar? Or we can work together to integrate it?
Other than that, just small questions from me; withapprove I will wait till we discuss and clean the hf repo; it's a mess, and I want to be sure we are all aligned on naming conventions (but its not related to your work, just a heads-up).

File: src/bert_dataset_generation/generate_coupon_selection_ds.py (hunk 3 above)
Comment (SzymonKozl) [reply src/bert_dataset_generation/generate_coupon_selection_ds.py:243]: This is the label associated with tokens that are not a part of coupon. In this context it means that the dataframe with no text is treated as there are no coupons in it. I have changed to `LBL_UNK` from `False` because this was a bug.

File: src/bert_dataset_generation/generate_coupon_selection_ds.py (hunk 1 above)
Comment (SzymonKozl) [reply src/bert_dataset_generation/generate_coupon_selection_ds.py:341]: At this moment I have decided to keep old functionality (merging all the frames into one dataset and then performing split between train, test and val) and add this new mode (saving each dataframe to the dedicated split) as optional. Should I remove the old mode completely?

File: src/bert_dataset_generation/readme.md (hunk 2 above)
Comment (SzymonKozl) [reply src/bert_dataset_generation/readme.md:9]: if custom_split=='y', then there is no traditional train-val-test split and each csv file is assigned to the split basing on the "split" fields in the config file.

File: src/bert_dataset_generation/generate_coupon_selection_ds.py (hunk 1 above)
Comment (GustawB) [reply src/bert_dataset_generation/generate_coupon_selection_ds.py:341]: I meant something else, but gi we will perform the split later anyway, so it can stay.

File: src/bert_dataset_generation/readme.md (hunk 2 above)
Comment (GustawB) [reply src/bert_dataset_generation/readme.md:9]: Ok, makes sense. Can you refraze it in the readme though? For now, it seems sta you set custom_split to 'n' nothing happens which is not true. I would also say into "what" splits will the data be divided (by that I mean each split represents data from a different app.)

Comment (SzymonKozl) [reply src/bert_dataset_generation/readme.md:9]: done

**Review by GustawB — APPROVED** [review]
